### DISCUSSION TITLE
某个日本 IP 段无法用于自选 IP [Error 1034]（Cloudflare CDN 刚刚添加的限制？）

### DISCUSSION THREAD

**Discussion by YujiaCheng1996** [XIU2/CloudflareSpeedTest#350] [讨论]
部分网站自选ip403，提示：

Please enable cookies.
Error 1034
Ray ID: xxxxx
Edge IP Restricted
What happened?
You've requested a page on a website that is part of the Cloudflare network. The host (xxx) resolved to an IP address that the owner of the website does not have access to.

What can I do?
If you are a visitor of this website:
Please try again in a few minutes.

If you are the owner of this website:
Check your DNS records to ensure they are pointed to the IP address(es) you were assigned at registration.

Cloudflare Ray ID: xxx • Your IP: xxx • Performance & security by Cloudflare

临时还是永久未可知。

**Comment by XIU2** [comment]
你也用的是我前段时间分享的那个**日本 IP 段**吧？
我换成其他自选 IP 就正常了。

提示内容翻译成人话大概意思是**该网站所有者没有权限使用该 IP**。

估计是该 IP 段**以前是专门用于** Cloudflare CDN 付费套餐的，然后发现不少人在用，于是加了限制。
或者说是该 IP 段**刚刚被划分为**专门用于 Cloudflare CDN 付费套餐。

**Comment by YujiaCheng1996** [comment]
除了正常通过dns解析出来的那3个ip，测试其它所有ip均403。
具体哪些网站和限制时长未可知。

**Comment by XIU2** [comment]
我是换成其他 IP 就正常了。

**Comment by XIU2** [comment]
“测试其它所有ip均403” 显然你不可能全部都测试过了，可能是你正好都碰上了被限制的 IP 段。

你把你测试用的 IP 段发给我看看。

**Comment by XIU2** [comment]
`104.16.248.130、104.16.249.130` ，这个是刚刚有个人在 discussions #74 分享的，可以正常使用。
`104.24.66.76` ，这个是我刚刚重新测速获得的最快 IP，依然可以正常使用，目前我只发现那个日本 IP 段无法使用了。

**Comment by YujiaCheng1996** [comment]
又测了下，是浏览器缓存没有更新。
172.64.228.x确定凉了，解析到其它暂没问题。
我的域名是Plesk Pro，开了Page Rules全缓存，并且使用官方dns，只是本地自选ip，自用的并没有很大流量，也没有超缓存大小限制。

**Comment by XIU2** [comment]
刚才我猜就可能是这样。。。

我这边浏览器不管是 <kbd>Ctrl+F5</kbd> 强制刷新（不走缓存），还是浏览器设置里清除 DNS 缓存，都不行。。。
必须重启浏览器才能完全清除 DNS 缓存（Hosts 才能生效），不过在用 3Proxy 后，就不存在这种问题了。

> 3Proxy 指的是：[还在一个个添加 Hosts？完美本地加速所有使用 Cloudflare CDN 的网站方法来了！](https://github.com/XIU2/CloudflareSpeedTest/discussions/71)

****

我猜测之所以只有这一个 IP 段被限制，很可能是因为该日本 IP 段，似乎**不是 Anycast 任播**的。
一直以来物理地址都是**日本东京**，我之前还很纳闷，怎么这个 **IP 段压根没变过**呢。。。

要知道其他的 IP 基本上**快则几个小时，慢则几天**就会 Anycast 动态调整一下，就这个日本 IP 段搞特殊化。。。

**Comment by YujiaCheng1996** [comment]
目前我是采用的本地脚本检测公网ip变化生成hosts，上传到我的网站，再在AdGuardHome或者设备中订阅这个hosts。
结果我这个网站选了这个ip段导致家里AdGuardHome获取不到dns上游的订阅规则断网了。。。
话说能不能在ip.txt把这个段排除了啊。

**Comment by XIU2** [comment]
目前不支持 IP 黑名单功能，即无法特定排除某个 IP / IP 段。

**Comment by XIU2** [comment]
但是，你可以选择把 `172.64.0.0/13` 这个 IP 段给拆分开，不写 `172.64.228.0/24` 就行了。

**Comment by YujiaCheng1996** [comment]
改CIDR呢？

**Comment by XIU2** [comment]
CIDR 无法精确的排除掉 172.64.228.0/24 这个 IP 段。

你只能拆分为一个个的 IP 段。

**Comment by YujiaCheng1996** [comment]
大佬，靠你了，写好更新下项目吧！

**Comment by XIU2** [comment]
把 `172.64.0.0/13` 拆分后就是这样子：

```
172.64.0.0/17
172.64.128.0/24
172.64.129.0/24
172.64.130.0/24
...
中间的自己写，漏掉 228 即可
...
172.64.254.0/24
172.64.255.0/24
172.65.0.0/16
172.66.0.0/16
172.67.0.0/16
172.68.0.0/16
172.69.0.0/16
172.70.0.0/16
172.71.0.0/16
```

**Comment by XIU2** [comment]
对了，你完全不用折腾这些，因为 `172.64.228.0/24` IP 段不可用的话，下载测速时会是 `0.00`（刚才还特地测试了下，确实是 0.00），这样你只要没有禁用下载测速（ `-dd` ），那该 IP 段就会被排除掉，或者你加上下载速度下限 `-sl 0.01` 也可以排除掉不可用的 IP（速度低于 0.01 MB/s 的 IP）。

**Comment by YujiaCheng1996** [comment]
```
for i in `seq 128 255`
do
        echo "172.64.$i.0/24"
done
```
228已删，更新下项目吧：
172.64.128.0/24
172.64.129.0/24
172.64.130.0/24
172.64.131.0/24
172.64.132.0/24
172.64.133.0/24
172.64.134.0/24
172.64.135.0/24
172.64.136.0/24
172.64.137.0/24
172.64.138.0/24
172.64.139.0/24
172.64.140.0/24
172.64.141.0/24
172.64.142.0/24
172.64.143.0/24
172.64.144.0/24
172.64.145.0/24
172.64.146.0/24
172.64.147.0/24
172.64.148.0/24
172.64.149.0/24
172.64.150.0/24
172.64.151.0/24
172.64.152.0/24
172.64.153.0/24
172.64.154.0/24
172.64.155.0/24
172.64.156.0/24
172.64.157.0/24
172.64.158.0/24
172.64.159.0/24
172.64.160.0/24
172.64.161.0/24
172.64.162.0/24
172.64.163.0/24
172.64.164.0/24
172.64.165.0/24
172.64.166.0/24
172.64.167.0/24
172.64.168.0/24
172.64.169.0/24
172.64.170.0/24
172.64.171.0/24
172.64.172.0/24
172.64.173.0/24
172.64.174.0/24
172.64.175.0/24
172.64.176.0/24
172.64.177.0/24
172.64.178.0/24
172.64.179.0/24
172.64.180.0/24
172.64.181.0/24
172.64.182.0/24
172.64.183.0/24
172.64.184.0/24
172.64.185.0/24
172.64.186.0/24
172.64.187.0/24
172.64.188.0/24
172.64.189.0/24
172.64.190.0/24
172.64.191.0/24
172.64.192.0/24
172.64.193.0/24
172.64.194.0/24
172.64.195.0/24
172.64.196.0/24
172.64.197.0/24
172.64.198.0/24
172.64.199.0/24
172.64.200.0/24
172.64.201.0/24
172.64.202.0/24
172.64.203.0/24
172.64.204.0/24
172.64.205.0/24
172.64.206.0/24
172.64.207.0/24
172.64.208.0/24
172.64.209.0/24
172.64.210.0/24
172.64.211.0/24
172.64.212.0/24
172.64.213.0/24
172.64.214.0/24
172.64.215.0/24
172.64.216.0/24
172.64.217.0/24
172.64.218.0/24
172.64.219.0/24
172.64.220.0/24
172.64.221.0/24
172.64.222.0/24
172.64.223.0/24
172.64.224.0/24
172.64.225.0/24
172.64.226.0/24
172.64.227.0/24
172.64.229.0/24
172.64.230.0/24
172.64.231.0/24
172.64.232.0/24
172.64.233.0/24
172.64.234.0/24
172.64.235.0/24
172.64.236.0/24
172.64.237.0/24
172.64.238.0/24
172.64.239.0/24
172.64.240.0/24
172.64.241.0/24
172.64.242.0/24
172.64.243.0/24
172.64.244.0/24
172.64.245.0/24
172.64.246.0/24
172.64.247.0/24
172.64.248.0/24
172.64.249.0/24
172.64.250.0/24
172.64.251.0/24
172.64.252.0/24
172.64.253.0/24
172.64.254.0/24
172.64.255.0/24

**Comment by XIU2** [comment]
你没看我前面说的么，没有必要改 `ip.txt` 了，因为这个 IP 段根本无法下载测速，结果只能是 0.00，会被排除掉。

实际上目前的 ip.txt 中并不是所有 IP 都能用的，其中有一部分的**回源 IP**（即是 CDN 去回源访问网站源服务器用的 IP），回源 IP 是不能用来指向域名访问的（下载测速也都是 0.00），剩下的就是任播 IP 才能这样用。

但是因为回源 IP 似乎**不是固定**的，所以我也没有在 `ip.txt` 中排除回源 IP，毕竟回源 IP 下载测速都是 0.00。

**Comment by YujiaCheng1996** [comment]
> 对了，你完全不用折腾这些，因为 `172.64.228.0/24` IP 段不可用的话，下载测速时会是 `0.00`（刚才还特地测试了下，确实是 0.00），这样你只要没有禁用下载测速（ `-dd` ），那该 IP 段就会被排除掉，或者你加上下载速度下限 `-sl 0.01` 也可以排除掉不可用的 IP（速度低于 0.01 MB/s 的 IP）。

我用的测速url不会被限速，workers.dev不限制，我的用加速github的workers测速，万一有人也使用的高级套餐的域名测速就会污染其它普通域名。

**Comment by XIU2** [comment]
这种情况，让用户自己处理吧。

毕竟绝大部分用户，都是用的 CloudflareST 默认的下载测速地址。

**Comment by YujiaCheng1996** [comment]
OK

**Comment by XIU2** [comment]
我后续没事的时候，会去扫扫**回源 IP 段**，看是不是固定的，如果是的话就给剔除掉算了。

这样的话一些人只在意延迟的话也不用担心遇到**不可用的回源 IP** 了。

**Comment by XIU2** [comment]
@YujiaCheng1996 对了，前面那个 IP 段可以精简一下：
```
172.64.0.0/17
172.64.128.0/18
172.64.192.0/19
172.64.224.0/22
172.64.229.0/24
172.64.230.0/23
172.64.232.0/21
172.64.240.0/21
172.64.248.0/21
```

这样的话，依然是正好跳过 `172.64.228.0/24` IP 段。

**Comment by XIU2** [comment]
@YujiaCheng1996 简单过滤了一下回源 IP（准确说是无法下载测速的 IP，即无法指向域名使用），不过因为 IP 太多（**约178万个**），所以也只是大概过滤了下。**大部分都过滤掉了**，应该还剩的有一部分（这部分混杂在各个 IP 段里，毫无规律，不确定是不是动态调整的），过滤掉的 IP 占原来的 **27.14%**。

> 过滤过程中发现，这部分 IP 绝大部分都是 80、443 端口超时，因此这些 IP 要么是被墙了、要么是本身闲置未使用的、要么就是回源 IP 了，不过其中还有部分 IP 80、443 端口是通顺的，但是无法下载测速（即无法指向域名）。搞不明白。 

下面是我过滤后得到的 `ip.txt` ，但是目前还不清楚回源 IP 是否会动态调整，所以也还没有同步到项目中。

```
1.1.1.0/24
1.0.0.0/24
1.1.1.1/32
1.0.0.1/32
173.245.49.0/24
103.21.244.0/24
141.101.64.0/18
190.93.240.0/20
198.41.128.0/17
162.159.0.0/16
104.16.0.0/12
172.64.0.0/17
172.64.128.0/18
172.64.192.0/19
172.64.224.0/22
172.64.229.0/24
172.64.230.0/23
172.64.232.0/21
172.64.240.0/21
172.64.248.0/21
172.65.49.0/24
172.65.60.0/24
172.65.217.0/24
172.65.227.0/24
172.65.240.0/24
172.65.253.0/24
172.67.0.0/16
```

**Comment by YujiaCheng1996** [comment]
感谢，我先手动替换着看看。

**Comment by Yzzzed** [comment]
用了3个网络去测试 通过3proxy去做代理 配合浏览器switchyOmega去测试 全都是Error 1034 是不是ipv4都不行了...
